src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/WindowsFormsAddItemFilter.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.WindowsForms
+{
+    /// <summary>
+    /// A filter for the Add New Item dialog that filters out Windows Forms items from non-Windows Forms projects.
+    /// </summary>
+    [ExportProjectNodeComService(typeof(IVsFilterAddProjectItemDlg))]
+    [AppliesTo("!" + ProjectCapability.WindowsForms)]
```

Comment:
This is too broad and will affect every single project-system, please combine _DotNet_.

Reply:
@davidwengier I just realized this won't work - "WinForms" is a dynamic capability (comes from targets) and these services need to be fixed for the lifetime of the project. This is enforced and will result in assertions/crashes in certain situations. This needs to be "DotNet" so that it is always present - and then you need to consult the project for the capabilities when the method is called.

Reply:
>  then you need to consult the project for the capabilities when the method is called

Yep, makes sense based on your other comment too. This is how I had it yesterday and then realized I can clean it up.. guess not :D

Reply:
Added a test to catch this: https://github.com/dotnet/project-system/pull/5520.